File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        for (int i=0;i<numbers.length;i++){
            int ind=searchInd(numbers,i+1,target-numbers[i]);
            if (ind!=-1){
                return new int[]{i+1,ind+1};
            }
        }
        return new int[]{-1,-1};
    }


    static int searchInd(int[]nums,int l,int tar){
        int r = nums.length - 1;

        while (l <= r) {
            int mid = (l + r) / 2;

            if (nums[mid] == tar) {
                return mid;
            } else if (nums[mid] < tar) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }

        return -1;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], tar: int) -> List[int]:
        i=0
        j=len(nums)-1
        while i<j:
            sum=nums[i]+nums[j]
            if sum==tar:
                return list([i+1,j+1])
            elif sum>tar:
                j-=1
            else:
                i+=1
